web/pages/index.tsx

```diff
@@ -37,30 +39,38 @@ import PostCreationModal from "../components/PostCreationModal/PostCreationModal
 import Feed from "../components/Feed/Feed";
 
 function Home() {
-  const { loading, authorized } = useAuth();
+  let { loading, authorized } = useAuth();
+
+  const [isLoading, setIsLoading] = useState(false);
   const { isOpen, onOpen, onClose } = useDisclosure();
 
+  const [alertDisplayed, setAlertDisplayed] = useState(false);
+
   async function handleLoginFacebook() {
     const provider = new FacebookAuthProvider();
-    await signInWithPopup(auth, provider);
+    try {
+      await signInWithPopup(auth, provider);
+    } catch (e) {
+      setIsLoading(false);
+      setAlertDisplayed(true);
+    }
   }
 
   async function handleLoginGoogle() {
     const provider = new GoogleAuthProvider();
-    await signInWithPopup(auth, provider);
+    try {
+      await signInWithPopup(auth, provider);
+    } catch (e) {
+      setIsLoading(false);
+      setAlertDisplayed(true);
+    }
```

Comment:
On error, display a toast with an error message provided by Firebase, see https://firebase.google.com/docs/auth/web/google-signin. I think they give you a status code and an error message